### ISSUE TITLE
setErrorReportingLevel not working with log handlers

### ISSUE THREAD

**Issue by amsoell** [bugsnag/bugsnag-laravel#245]
We're having an issue with a Laravel 5.4 project with Bugsnag integration where the `setErrorReportingLevel` method isn't properly filtering our messages sent to Bugsnag. We put the following in our `AppServiceProvder.php` file in the `boot()` method:

```
Bugsnag::setErrorReportingLevel(E_ALL & ~E_NOTICE);
```

Even after this, our notice-level log messages are still being pushed to Bugsnag. This line specifically still gets pushed out to our dashboard:

```
Log::notice('notice message');
```

Bugsnag technical support hasn't been very responsive, and I feel like we must be missing something obvious here. Any help is greatly appreciated

**Comment by kattrali** [comment]
Thanks for the report, @amsoell. It looks like the log handler doesn't use the error code filtering built into the error handler.

* [`shouldIgnoreErrorCode` logic](https://github.com/bugsnag/bugsnag-php/blob/master/src/Configuration.php#L481)
* [where the `errorReportingLevel` is used, in `Handler::errorHandler`](https://github.com/bugsnag/bugsnag-php/blob/master/src/Handler.php#L165)
* [`log` method in `BugsnagLogger`, which does not use `errorReportingLevel`/`errorHandler`](https://github.com/bugsnag/bugsnag-psr-logger/blob/master/src/BugsnagLogger.php#L40)

I think what we'd need to do to make this work would be one of a few options:

* bugsnag/bugsnag-psr-logger: Convert the string log level passed to `BugsnagHandler::log` to an [error reporting constant](http://php.net/manual/en/errorfunc.constants.php), and check `client->shouldIgnoreErrorCode()` before posting a log message
* bugsnag/bugsnag-laravel: Override `LaravelLogger::log` to do the same as the above before invoking parent.
* Create a new middleware for intercepting/filtering reports sent as log messages

What do you think, @GrahamCampbell? We should support a way to filter these, as the log level which created a report isn't clear by the time callbacks are invoked. I may also be missing something here. 🤔

**Comment by GrahamCampbell** [comment]
I think this is actually a documentation bug. The error level there is only respected by Bugsnag's Handler class (which is not used by Bugsnag Laravel), and not by the client object. It's just put in that main config object for ease of use, but is actually unused by Laravel. It should be removed from the Bugsnag Laravel docs if it's in there I'd say.

**Comment by kattrali** [comment]
Agreed, though it has come up a few times that there isn't an apparent way to filter by log level and that's something we could provide. My thought was that we could expose a minimum log level config option, but where that would belong would be up for discussion.

**Comment by kattrali** [comment]
This was actually fixed back in v2.10 using [`BUGSNAG_LOGGER_LEVEL`](https://docs.bugsnag.com/platforms/php/laravel/configuration-options/#logger-notify-level)
